plugins/Synchro/SynchroSynth.h

```diff
@@ -0,0 +1,153 @@
+/*
+ * SynchroSynth.h - 2-oscillator PM synth
+ *
+ * Copyright (c) 2020 Ian Sannar <ian/dot/sannar/at/gmail/dot/com>
+ * Credits to @DouglasDGI "Lost Robot" for performance optimizations
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+#ifndef SYNCHROSYNTH_H
+#define SYNCHROSYNTH_H
+
+#include "Graph.h" //Required for the waveform view
+#include "Instrument.h" //Required for the synth
+#include "InstrumentView.h" //Required for instrument GUI
+#include "Knob.h" //Required to use knobs
+#include "NotePlayHandle.h" //Required for audio rendering
+
+constexpr char SYNCHRO_VERSION [] = "0.6";
+constexpr float SYNCHRO_VOLUME_CONST = 0.15f; //Prevents clipping
+constexpr int SYNCHRO_OVERSAMPLE = 4; //Anti-aliasing samples
+constexpr int SYNCHRO_PM_CONST = 20; //Strength of the phase modulation
+constexpr int SYNCHRO_GRAPH_SAMPLES = 48; //Resolution of the waveform view
+constexpr float MAGIC_HARMONICS[2][2] = {{32, 0.5}, {38, 0.025}}; //Yoioioi
```

Comment:
I don't remember my exact methodology, but those harmonics (applied to the modulator in particular) give a certain sound that I had envisioned Synchro being able to make since I had the idea years ago.

I think I found those exact values by messing around with Sytrus in the FL Studio demo until I found the desired sound, then looked at exactly what harmonics I had ended up with. There were more than two harmonics, but I found that adding more than the first two had diminishing returns and so it would be easier to only worry about those two.

I'm not attached to that line of code in particular, just the sound it enables. If you've got a way to do it that's more readable/more proper, I'm all ears.